src/utils/daos_hdlr.c

```diff
@@ -2496,6 +2497,8 @@ fs_copy(struct file_dfs *src_file_dfs,
 				 * fail otherwise
 				 */
 				if (rc != EEXIST && rc != 0) {
+					D_FREE(next_path);
```

Comment:
this is fine, but maybe better to move them to the bottom (after out:) to avoid repeating calling free on those two? would need to remove where they are also freed though in the success path in the body function.

Reply:
Yeah, I went back and forth on this one. Mostly because the success path is in a loop. If we move the free to after out: then for the success path we will be calling free on something we know we've already free'ed previously

Reply:
ah i see. i didn't realize there was a loop there.